# Анализ статистики курса "Python. Анализ данных в маркетинге"

In [1]:
import pandas as pd #загрузка библиотеки для работы с данными

In [2]:
data_t = pd.read_excel('Python. АДвМ Задания.xlsx') #создание датасета с данными о прохождении практических заданий
data_test = pd.read_excel('Python. АДвМ КР.xlsx') #создание датасета с данными о прохождении контрольных работ

## Подсчет студентов, кто прошёл на 100 баллов 65% заданий

In [3]:
data_t.head() #отображение первых 5 строк датасета с заданиями

Фамилия                    Имя         Адрес электронной почты  \
0                -        Фоун Мьять Мин            foun.mm@edu.spbstu.ru   
1  Абдиллахи Барре                Лукман   abdillahibarre.l@edu.spbstu.ru   
2        Абдуллина     Елизавета Петровна      abdullina.ep@edu.spbstu.ru   
3          Абелева    Светлана Дмитриевна        abeleva.sd@edu.spbstu.ru   
4          Аборнев  Ярослав Владиславович       abornev.yav@edu.spbstu.ru   

  Виртуальная лаборатория программирования:Практическое задание 2 - Тренажер (Значение)  \
0                                                  0                                      
1                                                100                                      
2                                                100                                      
3                                                100                                      
4                                                  -                                      

  Виртуальная лаборатория программирования:Задание по NumPy 1 (обязательно) (Значение)  \
0                                                  -                                     
1                                                  -                                     
2                                                  -                                     
3                                                  -                                     
4                                                  -                                     

  Виртуальная лаборатория программирования:Практическое задание 2.1 (Значение)  \
0                                                  -                             
1                                                100                             
2                                                100                             
3                                                100                             
4                                                100                             

  Виртуальная лаборатория программирования:Практическое задание 2.4 (Значение)  \
0                                                  -                             
1                                                100                             
2                                                100                             
3                                                100                             
4                                                100                             

  Виртуальная лаборатория программирования:Практическое задание 2.2 (Значение)  \
0                                                  -                             
1                                                100                             
2                                                100                             
3                                                100                             
4                                                100                             

  Виртуальная лаборатория программирования:Практическое задание 2.3 (Значение)  \
0                                                  -                             
1                                                100                             
2                                                100                             
3                                                100                             
4                                                100                             

  Виртуальная лаборатория программирования:Практическое задание 2.5 (Значение)  \
0                                                  -                             
1                                                100                             
2                                                100                             
3                                                100                             
4                                                100                          

In [4]:
data_t.columns = data_t.columns.str.replace('Виртуальная лаборатория программирования:', '')
#коррекция названий столбцов для более удобной работы с ними
data_t.columns = data_t.columns.str.replace(' (Значение)', '', regex=False)

In [5]:
data_t.info() #общая информация о данных в датасете

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 687 entries, 0 to 686
Data columns (total 88 columns):
 #   Column                                             Non-Null Count  Dtype 
---  ------                                             --------------  ----- 
 0   Фамилия                                            687 non-null    object
 1   Имя                                                687 non-null    object
 2   Адрес электронной почты                            687 non-null    object
 3   Практическое задание 2 - Тренажер                  687 non-null    object
 4   Задание по NumPy 1 (обязательно)                   687 non-null    object
 5   Практическое задание 2.1                           687 non-null    object
 6   Практическое задание 2.4                           687 non-null    object
 7   Практическое задание 2.2                           687 non-null    object
 8   Практическое задание 2.3                           687 non-null    object
 9   Практическое задание 

In [6]:
data_tf = data_t.copy() #создание копии датасета для дальнейшей работы
for i in data_tf.columns: #цикл, который проходит по всем столбцам датасета
    data_tf.loc[data_tf[i] == '-', [i]] = None #замена всех знаков дефис, поскольку они не поддаются обработке
data_tf = data_tf.dropna(axis=1, how='all') #удаление пустых столбцов, которые были случайно загружены вместе с другими
data_tf_2 = data_tf.copy() #копирование датасета для дополительного анализа

In [7]:
data_tf.head() #первые 5 строк форматированного датасета

Фамилия                    Имя         Адрес электронной почты  \
0             None        Фоун Мьять Мин            foun.mm@edu.spbstu.ru   
1  Абдиллахи Барре                Лукман   abdillahibarre.l@edu.spbstu.ru   
2        Абдуллина     Елизавета Петровна      abdullina.ep@edu.spbstu.ru   
3          Абелева    Светлана Дмитриевна        abeleva.sd@edu.spbstu.ru   
4          Аборнев  Ярослав Владиславович       abornev.yav@edu.spbstu.ru   

  Практическое задание 2 - Тренажер Практическое задание 2.1  \
0                                 0                     None   
1                               100                      100   
2                               100                      100   
3                               100                      100   
4                              None                      100   

  Практическое задание 2.4 Практическое задание 2.2 Практическое задание 2.3  \
0                     None                     None                     None   
1                      100                      100                      100   
2                      100                      100                      100   
3                      100                      100                      100   
4                      100                      100                      100   

  Практическое задание 2.5 Практическое задание 2.6  ...  \
0                     None                     None  ...   
1                      100                      100  ...   
2                      100                      100  ...   
3                      100                      100  ...   
4                      100                      100  ...   

  Практическое задание pandas_3 Практическое задание pandas_4  \
0                          None                          None   
1                          None                          None   
2                          None                          None   
3                          None                          None   
4                          None                          None   

  Практическое задание pandas_5 Практическое задание pandas_6  \
0                          None                          None   
1                          None                          None   
2                          None                          None   
3                          None                          None   
4                          None                          None   

  Практическое задание pandas_7 Практическое задание pandas_8  \
0                          None                          None   
1                          None                          None   
2                          None                          None   
3                          None                          None   
4                          None                          None   

  Практическое задание pandas_9 Практическое задание pandas_10  \
0                          None                           None   
1                          None                           None   
2                          None                           None   
3                          None                           None   
4                          None                           None   

  Практическое задание pandas_11 (дополнительно)  \
0                                           None   
1                                           None   
2                                           None   
3                                           None   
4                                           None   

  Практическое задание pandas_12 (дополнительно)  
0                                           None  
1                                           None  
2                                           None  
3                                           None  
4                                           None  

[5 rows x 80 columns]

In [8]:
cols = data_tf.columns[3:] #ограничим столбцы только теми, которые содержат оценки по заданиям

In [9]:
round(len(cols) * 0.65) #подсчёт числа заданий, которок необходимо завершить для преодоления порога в 65%

50

In [10]:
data_tf['completed_t'] = data_tf[cols].count(axis=1) 
#посчитаем для каждого студента число завершенных заданий
#вне зависимости от оценки

In [11]:
data_tf['success_t'] = data_tf[data_tf[cols] == 100].count(axis=1) #посчитаем число успешно сданных заданий
#т.е. сданных на 100 баллов

In [12]:
data_tf['perfect_t_percent'] = round((data_tf['success_t'] / 77) * 100)
#посчитаем процент успешно завершенных заданий от общего числа доступных заданий

In [13]:
data_tf = data_tf.query('success_t >= 50') #выберем только тех студентов, которые прошли порог в 65% успешно сданных заданий 

In [14]:
data_tf = (data_tf[['Фамилия', 'Имя', 'Адрес электронной почты', 'completed_t', 'success_t', 'perfect_t_percent']]
 .sort_values(by='perfect_t_percent',ascending=False)
 .reset_index(drop=True)) #пересоздание индексов для удобной работы
#отсортируем полученный датасет по убыванию процента успешно завершенных заданий

In [15]:
data_tf #полученные данные

Фамилия                   Имя         Адрес электронной почты  \
0      Саморукова   Анастасия Сергеевна    samorukova2.as@edu.spbstu.ru   
1           Попов     Никита Евгеньевич        hearer2459@edu.spbstu.ru   
2        Сатторов   Дилшод Чоршанбиевич      sattorov.dch@edu.spbstu.ru   
3            Швец        Юлия Сергеевна        shvets.yus@edu.spbstu.ru   
4          Имаева  Екатерина Аркадьевна         imaeva.ea@edu.spbstu.ru   
5        Десятков       Артем Романович      desyatkov.ar@edu.spbstu.ru   
6          Гресев     Андрей Аркадиевич        hearer2148@edu.spbstu.ru   
7         Штыкина  Екатерина Георгиевна        hearer2497@edu.spbstu.ru   
8        Бутакова   Светлана Дмитриевна       butakova.sd@edu.spbstu.ru   
9       Максатова      Алина Максатовна        hearer2121@edu.spbstu.ru   
10         Бутько      Вячеслав Юрьевич        hearer1795@edu.spbstu.ru   
11       Демочкин      Игорь Дмитриевич      demochkin.id@edu.spbstu.ru   
12      Никоненко    Анастасия Петровна      nikonenko.ap@edu.spbstu.ru   
13    Желябовский    Всеволод Сергеевич  zhelyabovskij.vs@edu.spbstu.ru   
14        Цепинов      Никита Антонович        hearer2643@edu.spbstu.ru   
15      Коваленко  Михаил Александрович     kovalenko2.ma@edu.spbstu.ru   
16      Дорофеева    Анастасия Игоревна      dorofeeva.ai@edu.spbstu.ru   
17         Мухина        Ульяна Юрьевна        muhina.uyu@edu.spbstu.ru   
18       Федорова     Любовь Алексеевна       fedorova.la@edu.spbstu.ru   
19       Кутькина   Влада Александровна        hearer2261@edu.spbstu.ru   
20          Щукин     Иван Владимирович        hearer2202@edu.spbstu.ru   
21   Мустафокулов     Маруф Холбой Угли   mustafokulov.mh@edu.spbstu.ru   
22       Малышева      Юлия Геннадьевна     malysheva.yug@edu.spbstu.ru   
23        Яковлев    Алексей Алексеевич       yakovlev.aa@edu.spbstu.ru   
24        Иванова   Анастасия Сергеевна       ivanova6.as@edu.spbstu.ru   
25        Залавин    Савелий Николаевич        zalavin.sn@edu.spbstu.ru   
26         Алисов         Иван Иванович        hearer2412@edu.spbstu.ru   
27       Архипова   Маргарита Андреевна       arhipova.ma@edu.spbstu.ru   
28         Шипова  Анастасия Дмитриевна        hearer2306@edu.spbstu.ru   
29      Домрачева   Юлия Константиновна    domracheva.yuk@edu.spbstu.ru   
30       Сгибнева        Анна Андреевна       sgibneva.aa@edu.spbstu.ru   
31        Архипов        Артем Олегович        arhipov.ao@edu.spbstu.ru   
32        Баулина        Алена Олеговна        hearer2559@edu.spbstu.ru   
33        Харенко  Елизавета Дмитриевна        harenko.ed@edu.spbstu.ru   
34          Сечко      Артём Кириллович        hearer2285@edu.spbstu.ru   
35        Вокуева    Анастасия Игоревна        vokueva.ai@edu.spbstu.ru   
36       Гомзяков      Герман Сергеевич      gomzyakov.gs@edu.spbstu.ru   
37      Реентович      Дарья Николаевна        hearer1824@edu.spbstu.ru   
38        Егорова       Майя Михайловна        egorova.mm@edu.spbstu.ru   
39      Карагодин      Денис Алексеевич        hearer2731@edu.spbstu.ru   
40       Кулакова  Александра Сергеевна        hearer2335@edu.spbstu.ru   
41         Алиева       Милана Рафиевна        hearer2355@edu.spbstu.ru   
42       Коломиец    Мария Всеволодовна      kolomiets.mv@edu.spbstu.ru   
43    Василевский        Ильяс Ривович.        hearer2665@edu.spbstu.ru   
44    Большуткина         Элина Юрьевна   bolshutkina.eyu@edu.spbstu.ru   
45    Сероштанова      Полина Андреевна   seroshtanova.pa@edu.spbstu.ru   
46        Жильцов   Игорь Александрович        hearer1867@edu.spbstu.ru   
47  Зайнчуковская    Дарья Вячеславовна        hearer2378@edu.spbstu.ru   

    completed_t  success_t  perfect_t_percent  
0            77         77              100.0  
1            75         72               94.0  
2            70         66               86.0  
3            66         66               86.0  
4            64         63               82.0  
5            62         62               81.0 

**Комментарий:** Мы получили список из 48 студентов, которые успешно (*на 100 баллов*) прошли не менее 65% всех доступных на платформе заданий. К личным данным студентов также прилагается статистика прохождения заданий: общее число завершенных заданий (*completed_t*), общее число заданий, завершенных на 100 баллов (*success_t*) и процент успешно завершенных заданий от общего числа доступных (*perfect_t_percent*).

In [16]:
cols_f = [] #создание пустого списка

In [17]:
for n in data_tf_2.columns: #цикл, который проходит по всем столбцам датасета
    if 'дополнительно' not in n: #проверка наличия надписи в названии столбцов
        cols_f.append(n) #сохраняем в новый списки только основные задания

In [18]:
data_tf_2 = data_tf_2[cols_f] #оставим в датасете только столбцы, с основными заданиями

In [19]:
round(len(cols_f[3:]) * 0.65) #подсчёт числа заданий, которок необходимо завершить для преодоления порога в 65%
#поскольку в данном датасете остались только основные задания, число заданий, составляющее порог заметно меньше предыдущего

34

In [20]:
data_tf_2.head() #первые 5 строк датасета

Фамилия                    Имя         Адрес электронной почты  \
0             None        Фоун Мьять Мин            foun.mm@edu.spbstu.ru   
1  Абдиллахи Барре                Лукман   abdillahibarre.l@edu.spbstu.ru   
2        Абдуллина     Елизавета Петровна      abdullina.ep@edu.spbstu.ru   
3          Абелева    Светлана Дмитриевна        abeleva.sd@edu.spbstu.ru   
4          Аборнев  Ярослав Владиславович       abornev.yav@edu.spbstu.ru   

  Практическое задание 2 - Тренажер Практическое задание 2.1  \
0                                 0                     None   
1                               100                      100   
2                               100                      100   
3                               100                      100   
4                              None                      100   

  Практическое задание 2.4 Практическое задание 2.2 Практическое задание 2.3  \
0                     None                     None                     None   
1                      100                      100                      100   
2                      100                      100                      100   
3                      100                      100                      100   
4                      100                      100                      100   

  Практическое задание 2.5 Практическое задание 2.6  ...  \
0                     None                     None  ...   
1                      100                      100  ...   
2                      100                      100  ...   
3                      100                      100  ...   
4                      100                      100  ...   

  Практическое задание pandas_1 Практическое задание pandas_2  \
0                          None                          None   
1                          None                          None   
2                          None                          None   
3                          None                          None   
4                          None                          None   

  Практическое задание pandas_3 Практическое задание pandas_4  \
0                          None                          None   
1                          None                          None   
2                          None                          None   
3                          None                          None   
4                          None                          None   

  Практическое задание pandas_5 Практическое задание pandas_6  \
0                          None                          None   
1                          None                          None   
2                          None                          None   
3                          None                          None   
4                          None                          None   

  Практическое задание pandas_7 Практическое задание pandas_8  \
0                          None                          None   
1                          None                          None   
2                          None                          None   
3                          None                          None   
4                          None                          None   

  Практическое задание pandas_9 Практическое задание pandas_10  
0                          None                           None  
1                          None                           None  
2                          None                           None  
3                          None                           None  
4                          None                           None  

[5 rows x 56 columns]

In [21]:
data_tf_2['completed_t'] = data_tf_2[cols_f[3:]].count(axis=1) #посчитаем для каждого студента число завершенных заданий
#вне зависимости от оценки

In [22]:
data_tf_2['success_t'] = data_tf_2[data_tf_2[cols_f[3:]] == 100].count(axis=1)#посчитаем число успешно сданных заданий
#т.е. сданных на 100 баллов

In [23]:
data_tf_2['perfect_t_percent'] = round((data_tf_2['success_t'] / 53) * 100)
#посчитаем процент успешно завершенных заданий от общего числа доступных заданий

In [24]:
data_tf_2 = data_tf_2.query('success_t > 34')
#выберем только тех студентов, которые прошли порог в 65% успешно сданных заданий 

In [25]:
data_tf_2 = (data_tf_2[['Фамилия', 'Имя', 'Адрес электронной почты', 'completed_t', 'success_t', 'perfect_t_percent']]
 .sort_values(by='perfect_t_percent',ascending=False)
 .reset_index(drop=True))#пересоздание индексов для удобной работы
#отсортируем полученный датасет по убыванию процента успешно завершенных заданий

In [26]:
data_tf_2 #полученный датасет

Фамилия                   Имя       Адрес электронной почты  \
0         Егорова       Майя Михайловна      egorova.mm@edu.spbstu.ru   
1         Архипов        Артем Олегович      arhipov.ao@edu.spbstu.ru   
2       Дорофеева    Анастасия Игоревна    dorofeeva.ai@edu.spbstu.ru   
3           Сечко      Артём Кириллович      hearer2285@edu.spbstu.ru   
4        Сгибнева        Анна Андреевна     sgibneva.aa@edu.spbstu.ru   
..            ...                   ...                           ...   
288     Бессолова       Полина Петровна    bessolova.pp@edu.spbstu.ru   
289      Толкачев    Дмитрий Васильевич    tolkachev.dv@edu.spbstu.ru   
290       Федоров  Святослав Дмитриевич      hearer2498@edu.spbstu.ru   
291  Хазиахметова               Гузаль   haziahmetova.g@edu.spbstu.ru   
292     Гасникова  Ксения Александровна    gasnikova.ka@edu.spbstu.ru   

     completed_t  success_t  perfect_t_percent  
0             53         53              100.0  
1             53         53              100.0  
2             53         53              100.0  
3             53         53              100.0  
4             53         53              100.0  
..           ...        ...                ...  
288           39         35               66.0  
289           35         35               66.0  
290           35         35               66.0  
291           35         35               66.0  
292           35         35               66.0  

[293 rows x 6 columns]

**Комментарий:** Полученный датасет содержит информацию о 293 студентах, которые прошли не менее 65% основных заданий на 100 баллов. Так, можно сказать, что чуть меньше половины всех студентов, успешно справились с основными заданиями.

## Подсчет студентов, кто прошёл на 65% контрольные вопросы по всем темам

In [27]:
data_test.info() #основная информация о датасете с контрольными заданиями

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 687 entries, 0 to 686
Data columns (total 12 columns):
 #   Column                                 Non-Null Count  Dtype 
---  ------                                 --------------  ----- 
 0   Фамилия                                687 non-null    object
 1   Имя                                    687 non-null    object
 2   Адрес электронной почты                687 non-null    object
 3   Тест:Контрольные вопросы (Значение)    687 non-null    object
 4   Тест:Контрольные вопросы (Значение).1  687 non-null    object
 5   Тест:Контрольные вопросы (Значение).2  687 non-null    object
 6   Тест:Контрольные вопросы (Значение).3  687 non-null    object
 7   Тест:Контрольные вопросы (Значение).4  687 non-null    object
 8   Тест:Контрольные вопросы (Значение).5  687 non-null    object
 9   Тест:Контрольные вопросы (Значение).6  687 non-null    object
 10  Тест:Контрольные вопросы (Значение).7  687 non-null    object
 11  Тест:Контрольные во

In [28]:
data_test.columns = data_test.columns.str.replace('Тест:', '').str.replace(' (Значение)', '', regex=False)
data_test.columns = data_test.columns.str.replace(' ', '_')
#коррекция назаваний столбцов для более удобной работы с ними

In [29]:
data_test.head() #первые 5 строк датасета

Фамилия                    Имя         Адрес_электронной_почты  \
0                -        Фоун Мьять Мин            foun.mm@edu.spbstu.ru   
1  Абдиллахи Барре                Лукман   abdillahibarre.l@edu.spbstu.ru   
2        Абдуллина     Елизавета Петровна      abdullina.ep@edu.spbstu.ru   
3          Абелева    Светлана Дмитриевна        abeleva.sd@edu.spbstu.ru   
4          Аборнев  Ярослав Владиславович       abornev.yav@edu.spbstu.ru   

  Контрольные_вопросы Контрольные_вопросы.1 Контрольные_вопросы.2  \
0                  10                     -                     -   
1                  10                     5                     5   
2                  10                     5                     5   
3                  10                     5                     5   
4                   -                     -                     -   

  Контрольные_вопросы.3 Контрольные_вопросы.4 Контрольные_вопросы.5  \
0                     -                     -                     -   
1                     5                     -                    10   
2                     5                     -                  8.89   
3                     5                     -                    10   
4                     -                     -                     -   

  Контрольные_вопросы.6 Контрольные_вопросы.7 Контрольные_вопросы.8  
0                     -                     -                     -  
1                    10                    10                    10  
2                     -                     -                     -  
3                     -                     -                     -  
4                     -                     -                     -

In [30]:
cols_test = data_test.columns #создание списка с названиями столбцов

In [31]:
for i in data_test.columns:
    data_test.loc[data_test[i] == '-', [i]] = None
#замена в цикле знаков дефис на None для обработки датасета

In [32]:
data_test = data_test.dropna(axis=1, how='all') #удалим пустые, столбцы, которые могли случайно попасть в выгрузку

In [33]:
cols_test = data_test.columns #новый список с названиями столбцов

In [34]:
data_test['completed_test'] = data_test[cols_test[3:]].count(axis=1) #подсчёт общего числа завершенных тестов
#вне зависимости от оценки

In [35]:
data_test[cols_test[3:]] = data_test[cols_test[3:]].fillna(0) #замена всех пропусков на 0,
#поскольку иначе невозможно посчитать оценки
data_test[cols_test[3:]] = data_test[cols_test[3:]].astype('float') #изменение типа данных в датасете на вещественный

In [36]:
data_test.head() #первые 5 строк полученного датасета

Фамилия                    Имя         Адрес_электронной_почты  \
0             None        Фоун Мьять Мин            foun.mm@edu.spbstu.ru   
1  Абдиллахи Барре                Лукман   abdillahibarre.l@edu.spbstu.ru   
2        Абдуллина     Елизавета Петровна      abdullina.ep@edu.spbstu.ru   
3          Абелева    Светлана Дмитриевна        abeleva.sd@edu.spbstu.ru   
4          Аборнев  Ярослав Владиславович       abornev.yav@edu.spbstu.ru   

   Контрольные_вопросы  Контрольные_вопросы.1  Контрольные_вопросы.2  \
0                 10.0                    0.0                    0.0   
1                 10.0                    5.0                    5.0   
2                 10.0                    5.0                    5.0   
3                 10.0                    5.0                    5.0   
4                  0.0                    0.0                    0.0   

   Контрольные_вопросы.3  Контрольные_вопросы.5  Контрольные_вопросы.6  \
0                    0.0                   0.00                    0.0   
1                    5.0                  10.00                   10.0   
2                    5.0                   8.89                    0.0   
3                    5.0                  10.00                    0.0   
4                    0.0                   0.00                    0.0   

   Контрольные_вопросы.7  Контрольные_вопросы.8  completed_test  
0                    0.0                    0.0               1  
1                   10.0                   10.0               8  
2                    0.0                    0.0               5  
3                    0.0                    0.0               5  
4                    0.0                    0.0               0

In [37]:
data_test['success_test_5'] = data_test[data_test[cols_test[4:7]] >= 3.25].count(axis=1)
#подсчёт успешно завершенных тестов с макс.оценкой в 5 баллов (65% правильных ответов - балл выше 3.25)
data_test['success_test_10'] = data_test[data_test[cols_test[[3,7,8,9,10]]] >= 6.5].count(axis=1)
#подсчёт успешно завершенных тестов с макс.оценкой в 10 баллов (65% правильных ответов - балл выше 6.5)
data_test['success_total'] = data_test['success_test_10'] + data_test['success_test_5']
#объединение полученных значений для подсчёта общего числа успешных тестов

In [38]:
data_test.head() #первые 5 строк полученного датасета

Фамилия                    Имя         Адрес_электронной_почты  \
0             None        Фоун Мьять Мин            foun.mm@edu.spbstu.ru   
1  Абдиллахи Барре                Лукман   abdillahibarre.l@edu.spbstu.ru   
2        Абдуллина     Елизавета Петровна      abdullina.ep@edu.spbstu.ru   
3          Абелева    Светлана Дмитриевна        abeleva.sd@edu.spbstu.ru   
4          Аборнев  Ярослав Владиславович       abornev.yav@edu.spbstu.ru   

   Контрольные_вопросы  Контрольные_вопросы.1  Контрольные_вопросы.2  \
0                 10.0                    0.0                    0.0   
1                 10.0                    5.0                    5.0   
2                 10.0                    5.0                    5.0   
3                 10.0                    5.0                    5.0   
4                  0.0                    0.0                    0.0   

   Контрольные_вопросы.3  Контрольные_вопросы.5  Контрольные_вопросы.6  \
0                    0.0                   0.00                    0.0   
1                    5.0                  10.00                   10.0   
2                    5.0                   8.89                    0.0   
3                    5.0                  10.00                    0.0   
4                    0.0                   0.00                    0.0   

   Контрольные_вопросы.7  Контрольные_вопросы.8  completed_test  \
0                    0.0                    0.0               1   
1                   10.0                   10.0               8   
2                    0.0                    0.0               5   
3                    0.0                    0.0               5   
4                    0.0                    0.0               0   

   success_test_5  success_test_10  success_total  
0               0                1              1  
1               3                5              8  
2               3                2              5  
3               3                2              5  
4               0                0              0

In [39]:
data_test_f = (data_test.query('success_total == 8')
               [['Фамилия', 'Имя', 'Адрес_электронной_почты', 'success_total']]
               .reset_index(drop=True)) #выборка из студентов, которые успешно сдали все тесты (оценка не менее 65%)

In [40]:
data_test_f #полученный датасет

Фамилия                     Имя         Адрес_электронной_почты  \
0   Абдиллахи Барре                 Лукман   abdillahibarre.l@edu.spbstu.ru   
1            Алисов           Иван Иванович        hearer2412@edu.spbstu.ru   
2           Архипов          Артем Олегович        arhipov.ao@edu.spbstu.ru   
3          Архипова     Маргарита Андреевна       arhipova.ma@edu.spbstu.ru   
4            Бруева     Анжелика Алексеевна        hearer2518@edu.spbstu.ru   
5         Воробьёва     Елизавета Андреевна     vorobieva3.ea@edu.spbstu.ru   
6             Гедзь     Екатерина Сергеевна        hearer2438@edu.spbstu.ru   
7          Гомзяков        Герман Сергеевич      gomzyakov.gs@edu.spbstu.ru   
8        Де Алмейда            Иван Анибал         hearer1841@edu.spbstu.ru   
9          Демочкин        Игорь Дмитриевич      demochkin.id@edu.spbstu.ru   
10        Дмитриева       Анастасия Юрьевна    kolesnikova.ad@edu.spbstu.ru   
11        Домрачева     Юлия Константиновна    domracheva.yuk@edu.spbstu.ru   
12          Егорова         Майя Михайловна        egorova.mm@edu.spbstu.ru   
13          Жильцов     Игорь Александрович        hearer1867@edu.spbstu.ru   
14          Иванова     Анастасия Сергеевна       ivanova6.as@edu.spbstu.ru   
15        Карагодин        Денис Алексеевич        hearer2731@edu.spbstu.ru   
16        Коваленко    Михаил Александрович     kovalenko2.ma@edu.spbstu.ru   
17         Кулакова    Александра Сергеевна        hearer2335@edu.spbstu.ru   
18               Ли  Виолетта Владиславовна            li2.vv@edu.spbstu.ru   
19     Мустафокулов       Маруф Холбой Угли   mustafokulov.mh@edu.spbstu.ru   
20           Мухина          Ульяна Юрьевна        muhina.uyu@edu.spbstu.ru   
21        Никоненко      Анастасия Петровна      nikonenko.ap@edu.spbstu.ru   
22       Озяныкоглу                  Фейза      ozyanykoglu.f@edu.spbstu.ru   
23    Подлобошников         Артём Сергеевич        hearer2234@edu.spbstu.ru   
24            Попов       Никита Евгеньевич        hearer2459@edu.spbstu.ru   
25        Реентович        Дарья Николаевна        hearer1824@edu.spbstu.ru   
26       Саморукова     Анастасия Сергеевна    samorukova2.as@edu.spbstu.ru   
27          Смирнов     Максим Владимирович       smirnov4.mv@edu.spbstu.ru   
28          Харенко    Елизавета Дмитриевна        harenko.ed@edu.spbstu.ru   
29             Чжао                    Фан           chzhao.f@edu.spbstu.ru   
30           Шипова    Анастасия Дмитриевна        hearer2306@edu.spbstu.ru   

    success_total  
0               8  
1               8  
2               8  
3               8  
4               8  
5               8  
6               8  
7               8  
8               8  
9               8  
10              8  
11              8  
12              8  
13              8  
14              8  
15              8  
16              8  
17              8  
18              8  
19              8  
20              8  
21              8  
22              8  
23              8  
24              8  
25              8  
26              8  
27              8  
28              8  
29              8  
30              8

**Комментарий:** В список вошло 30 студентов, которые успешно сдали все тесты (*success_total*)

In [41]:
fav = pd.merge(data_test_f,
               data_tf[['Фамилия', 'Имя', 'completed_t','success_t','perfect_t_percent']],
               how='inner',
               on=['Фамилия', 'Имя'])
#объединение датасета со списком студентов, успешно сдавших все практические задания
#и датасета со списком студентов, успешно сдавших все тесты

In [42]:
fav #полученный датасет

Фамилия                   Имя        Адрес_электронной_почты  \
0         Алисов         Иван Иванович       hearer2412@edu.spbstu.ru   
1        Архипов        Артем Олегович       arhipov.ao@edu.spbstu.ru   
2       Архипова   Маргарита Андреевна      arhipova.ma@edu.spbstu.ru   
3       Гомзяков      Герман Сергеевич     gomzyakov.gs@edu.spbstu.ru   
4       Демочкин      Игорь Дмитриевич     demochkin.id@edu.spbstu.ru   
5      Домрачева   Юлия Константиновна   domracheva.yuk@edu.spbstu.ru   
6        Егорова       Майя Михайловна       egorova.mm@edu.spbstu.ru   
7        Жильцов   Игорь Александрович       hearer1867@edu.spbstu.ru   
8        Иванова   Анастасия Сергеевна      ivanova6.as@edu.spbstu.ru   
9      Карагодин      Денис Алексеевич       hearer2731@edu.spbstu.ru   
10     Коваленко  Михаил Александрович    kovalenko2.ma@edu.spbstu.ru   
11      Кулакова  Александра Сергеевна       hearer2335@edu.spbstu.ru   
12  Мустафокулов     Маруф Холбой Угли  mustafokulov.mh@edu.spbstu.ru   
13        Мухина        Ульяна Юрьевна       muhina.uyu@edu.spbstu.ru   
14     Никоненко    Анастасия Петровна     nikonenko.ap@edu.spbstu.ru   
15         Попов     Никита Евгеньевич       hearer2459@edu.spbstu.ru   
16     Реентович      Дарья Николаевна       hearer1824@edu.spbstu.ru   
17    Саморукова   Анастасия Сергеевна   samorukova2.as@edu.spbstu.ru   
18       Харенко  Елизавета Дмитриевна       harenko.ed@edu.spbstu.ru   
19        Шипова  Анастасия Дмитриевна       hearer2306@edu.spbstu.ru   

    success_total  completed_t  success_t  perfect_t_percent  
0               8           54         54               70.0  
1               8           53         53               69.0  
2               8           54         54               70.0  
3               8           54         53               69.0  
4               8           59         59               77.0  
5               8           54         54               70.0  
6               8           54         53               69.0  
7               8           52         50               65.0  
8               8           55         55               71.0  
9               8           53         53               69.0  
10              8           60         58               75.0  
11              8           54         53               69.0  
12              8           59         55               71.0  
13              8           62         56               73.0  
14              8           62         59               77.0  
15              8           75         72               94.0  
16              8           53         53               69.0  
17              8           77         77              100.0  
18              8           53         53               69.0  
19              8           56         54               70.0

**Комментарий:** В данном списке находится 19 студентов, которые успешно завершили как все практические задания, так и все тесты.

In [43]:
fav_no_added = data_test_f.merge(data_tf_2[['Фамилия', 'Имя', 'completed_t','success_t','perfect_t_percent']],
                how='inner',
                on=['Фамилия', 'Имя'])
#объединение датасета со списком студентов, успешно сдавших только основные практические задания
#и датасета со списком студентов, успешно сдавших все тесты

In [44]:
fav_no_added #полученный датасет

Фамилия                     Имя         Адрес_электронной_почты  \
0   Абдиллахи Барре                 Лукман   abdillahibarre.l@edu.spbstu.ru   
1            Алисов           Иван Иванович        hearer2412@edu.spbstu.ru   
2           Архипов          Артем Олегович        arhipov.ao@edu.spbstu.ru   
3          Архипова     Маргарита Андреевна       arhipova.ma@edu.spbstu.ru   
4            Бруева     Анжелика Алексеевна        hearer2518@edu.spbstu.ru   
5         Воробьёва     Елизавета Андреевна     vorobieva3.ea@edu.spbstu.ru   
6             Гедзь     Екатерина Сергеевна        hearer2438@edu.spbstu.ru   
7          Гомзяков        Герман Сергеевич      gomzyakov.gs@edu.spbstu.ru   
8        Де Алмейда            Иван Анибал         hearer1841@edu.spbstu.ru   
9          Демочкин        Игорь Дмитриевич      demochkin.id@edu.spbstu.ru   
10        Дмитриева       Анастасия Юрьевна    kolesnikova.ad@edu.spbstu.ru   
11        Домрачева     Юлия Константиновна    domracheva.yuk@edu.spbstu.ru   
12          Егорова         Майя Михайловна        egorova.mm@edu.spbstu.ru   
13          Жильцов     Игорь Александрович        hearer1867@edu.spbstu.ru   
14          Иванова     Анастасия Сергеевна       ivanova6.as@edu.spbstu.ru   
15        Карагодин        Денис Алексеевич        hearer2731@edu.spbstu.ru   
16        Коваленко    Михаил Александрович     kovalenko2.ma@edu.spbstu.ru   
17         Кулакова    Александра Сергеевна        hearer2335@edu.spbstu.ru   
18               Ли  Виолетта Владиславовна            li2.vv@edu.spbstu.ru   
19     Мустафокулов       Маруф Холбой Угли   mustafokulov.mh@edu.spbstu.ru   
20           Мухина          Ульяна Юрьевна        muhina.uyu@edu.spbstu.ru   
21        Никоненко      Анастасия Петровна      nikonenko.ap@edu.spbstu.ru   
22            Попов       Никита Евгеньевич        hearer2459@edu.spbstu.ru   
23        Реентович        Дарья Николаевна        hearer1824@edu.spbstu.ru   
24       Саморукова     Анастасия Сергеевна    samorukova2.as@edu.spbstu.ru   
25          Смирнов     Максим Владимирович       smirnov4.mv@edu.spbstu.ru   
26          Харенко    Елизавета Дмитриевна        harenko.ed@edu.spbstu.ru   
27             Чжао                    Фан           chzhao.f@edu.spbstu.ru   
28           Шипова    Анастасия Дмитриевна        hearer2306@edu.spbstu.ru   

    success_total  completed_t  success_t  perfect_t_percent  
0               8           43         43               81.0  
1               8           53         53              100.0  
2               8           53         53              100.0  
3               8           53         53              100.0  
4               8           43         42               79.0  
5               8           49         42               79.0  
6               8           43         42               79.0  
7               8           53         52               98.0  
8               8           45         44               83.0  
9               8           45         45               85.0  
10              8           50         49               92.0  
11              8           53         53              100.0  
12              8           53         53              100.0  
13              8           51         49               92.0  
14              8           53         53              100.0  
15              8           53         53              100.0  
16              8           53         53              100.0  
17              8           53         53              100.0  
18              8           42         42               79.0  
19              8           43         43               81.0  
20              8           43         43               81.0  
21              8           53         53              100.0  
22              8           53         53              100.0  
23              8           53         53              100.0  
24              8           53         53              100.0  
25 

**Комментарий:** В данном списке содержится 28 студентов, которые успешно завершили все основные задания и сдали все контрольные работы.

In [45]:
fav.to_excel('Студенты, успешно сдавшие все задания и тесты.xlsx') #сохранение датасета в файл эксель

In [46]:
fav_no_added.to_excel('Студенты, успешно сдавшие только основные задания и тесты.xlsx') #сохранение датасета в файл эксель